setuptools/tests/config/test_setupcfg.py

```diff
@@ -716,6 +716,41 @@ def test_extras_require(self, tmpdir):
             }
             assert dist.metadata.provides_extras == set(['pdf', 'rest'])
 
+    @pytest.mark.parametrize(
+        "config",
+        [
+            "[options.extras_require]\nfoo = bar;python_version<'3'",
+            "[options.extras_require]\nfoo = bar;os_name=='linux'"
```

Comment:
It seems that the `fake_env` simply writes the given string as a `setup.cfg` file:

```python
config.write(setup_cfg.encode(encoding), mode='wb')
```

My question here is: what happens if the text file ends in `\n`? (all text files in POSIX systems should end in a `\n` character right?)

Is it worthy to change the examples to (or add them as extra examples):
```python
"[options.extras_require]\nfoo = bar;python_version<'3'\n",
"[options.extras_require]\nfoo = bar;os_name=='linux'\n"
```

Should we also add something like the following to the list of examples? (In theory, `ConfigParser` should discard that line, so when it reaches `setuptools` the comment is already not an issue. I am not sure if it is worthy adding to the test).

```ini
[options.extras_require]
foo = bar; python_version<'3'
      # I am a comment
```

Reply:
Will add some more tests as suggested.